src/Compilers/Core/Portable/Syntax/InternalSyntaxNodeExtensions.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal static class InternalSyntaxNodeExtensions
+    {
+        /// <summary>
+        /// Gets the first node of type TNode that matches the predicate.
+        /// </summary>
+        internal static TNode? FirstAncestorOrSelf<TNode, TArg>(this SyntaxNode? node, Func<TNode, TArg, bool> predicate, TArg argument, bool ascendOutOfTrivia = true)
```

Comment:
>FirstAncestorOrSelf [](start = 31, length = 19)

Can we place this method next to the other ```FirstAncestorOrSelf```?

Reply:
Only if we also do one of the following:

* Make the method public
* Duplicate the method into the Workspaces layer

Reply:
> Only if we also do one of the following: ...

It is not clear what question is being answered here.

---
In reply to: [362616736](https://github.com/dotnet/roslyn/pull/40685#discussion_r362616736) [](ancestors = 362616736)

Reply:
This change was removed from the pull request.